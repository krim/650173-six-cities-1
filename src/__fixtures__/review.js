import user from './user';

export default {
  id: 1,
  user,
  rating: 3,
  comment: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
  date: `2019-05-21T05:25:32.222Z`
};
