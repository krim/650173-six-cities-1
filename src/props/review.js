import PropTypes from 'prop-types';
import userProps from './user';

export default PropTypes.shape({
  id: PropTypes.number.isRequired,
  user: userProps,
  rating: PropTypes.number.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired
});
