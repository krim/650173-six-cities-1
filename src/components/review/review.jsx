import React, {PureComponent} from 'react';

import {reviewProps} from '../../props';
import getRating from '../../libs/get-rating';

class Review extends PureComponent {
  render() {
    const {review} = this.props;
    const {user} = review;

    return (
      <React.Fragment>
        <div className="reviews__user user">
          <div className="reviews__avatar-wrapper user__avatar-wrapper">
            <img
              className="reviews__avatar user__avatar"
              src={user.avatarUrl}
              width="54"
              height="54"
              alt="Reviews avatar"
            />
          </div>
          <span className="reviews__user-name">
            { user.name }
          </span>
        </div>
        <div className="reviews__info">
          <div className="reviews__rating rating">
            <div className="reviews__stars rating__stars">
              <span style={{width: getRating(review.rating)}}></span>
              <span className="visually-hidden">Rating</span>
            </div>
          </div>
          <p className="reviews__text">
            {review.comment}
          </p>
          <time className="reviews__time" dateTime={this._getFormatedDateTime()}>{this._getFormatedDate()}</time>
        </div>
      </React.Fragment>
    );
  }

  _getFormatedDateTime() {
    const date = this._getReviewDate();
    const options = {year: `numeric`, month: `2-digit`, day: `2-digit`};
    return new Date(date).toLocaleString(`se-SE`, options);
  }

  _getFormatedDate() {
    const date = this._getReviewDate();
    const options = {year: `numeric`, month: `long`};
    return new Date(date).toLocaleString(`en-US`, options);
  }

  _getReviewDate() {
    return Date.parse(this.props.review.date);
  }
}

Review.propTypes = {
  review: reviewProps
};

export default Review;

