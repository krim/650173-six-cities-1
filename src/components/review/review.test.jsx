import React from 'react';
import renderer from 'react-test-renderer';

import Review from './review';
import review from './../../__fixtures__/review';

describe(`Review`, () => {
  it(`renders component correctly`, () => {
    const tree = renderer.
      create(<Review
        review={review}
      />).
      toJSON();

    expect(tree).toMatchSnapshot();
  });
});
