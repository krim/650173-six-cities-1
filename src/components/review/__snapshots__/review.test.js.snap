// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders component correctly 1`] = `
Array [
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="https://es31-server.appspot.com/six-cities/static/avatar/4.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Zak
    </span>
  </div>,
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "60%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
    </p>
    <time
      className="reviews__time"
      dateTime="2019-05-21"
    >
      May 2019
    </time>
  </div>,
]
`;
